src/teleop.rs

```diff
@@ -0,0 +1,75 @@
+use crate::config::TeleopConfig;
+use rosrust;
+use rosrust_msg;
+use std::collections::HashMap;
+use termion::event::Key;
+
+pub struct Teleoperator {
+    current_velocities: Velocities,
+    _cmd_val_pub: rosrust::Publisher<rosrust_msg::geometry_msgs::Twist>,
+    _keys_to_direction: HashMap<Key, (String, i8)>,
+    _increment: f64,
+}
+
+pub struct Velocities {
+    x: f64,
+    y: f64,
+    theta: f64,
+}
+
+fn match_key_to_string(
+    string_to_direction: &HashMap<String, (String, i8)>,
+) -> HashMap<Key, (String, i8)> {
+    let mut keys_to_direction = HashMap::new();
+    for (string, val) in string_to_direction {
```

Comment:
Why do you think that's better?
I find it easy to understand and performance wise its not relevant.

Reply:
Because you are writing 5 lines and creating a support mutable variable when the entire match_key_to_string function is simply:
```
string_to_direction.iter().map(|(k,v)| (Key::Char(k.chars().next().unwrap()), v.clone())).collect()
```